File: packages/core/src/types/registry.ts
import { GlobalConfig } from "domain/entity/GlobalConfig";
import { WorkerClientFactory } from "domain/factory/WorkerClientFactory";
import { PluginRepository } from "domain/repository/PluginRepository";
import { WorkerRepository } from "domain/repository/WorkerRepository";
import { TemporaryStoreRepository } from "domain/repository/TemporaryStoreRepository";

export type Registry = {
	workerClientFactory: WorkerClientFactory;
	workerRepository: WorkerRepository;
	globalConfig: GlobalConfig;
	pluginRepository: PluginRepository;
	temporaryStoreRepository: TemporaryStoreRepository;
};

File: packages/worker/types/registry.go
package types

import (
	"github.com/kk-mats/ccx-worker/domain/object/value"
	"github.com/kk-mats/ccx-worker/domain/repository/client"
	"github.com/kk-mats/ccx-worker/domain/repository/filesystem"
	"github.com/kk-mats/ccx-worker/domain/service"
)

type Service struct {
	Detect *service.Detect
}

type Repository struct {
	WorkspaceFilesystem filesystem.Workspace
	PluginFileSystem    filesystem.Plugin
	PrimeClient         client.Prime
	ContainerClient     client.Container
}

type Registry struct {
	GlobalConfig *value.GlobalConfig
	Service      *Service
	Repository   *Repository
}
